src/file_upload.rs

```diff
@@ -69,6 +70,17 @@ fn handle_multipart(
     let err = |e: ContextualError| Box::pin(future::err(e).into_stream());
     match filename {
         Ok(f) => {
+            // Without this check, attackers can mount an attack via path traversal by
+            // manipulating the filename attribute in the multipart request in the
+            // ways enumerated on this page: https://cwe.mitre.org/data/definitions/22.html
+            match &app_root_dir.join(&f).canonicalize() {
+                Ok(path) if path.starts_with(&app_root_dir) => (),
```

Comment:
```suggestion
            match &file_path.join(&f).canonicalize() {
                Ok(path) if path.starts_with(&file_path) => (),
```

No need for `app_root_dir`. Ideally, I think, we should have something similar to `sanitize_path` in #476 .

Reply:
Thanks!

My code joining the filename to the app_root_dir is just wrong and would put files in the wrong place if users were trying to upload to a subfolder.

I'll also check out your suggestion on sanitize_path later today.

Reply:
You may leave it as it is, just fix `app_root_dir` issue and I'll polish it in my open PR.

Reply:
Awesome!

I made the change in the latest commit.